### DISCUSSION TITLE
Add to node embeddings in a dynamic fashon in GCNConv

### DISCUSSION THREAD

**Discussion by Sanjaykrishnamurthy** [pyg-team/pytorch_geometric#7686] [Q&A] [answered]
I have the nodes and their temporal properties like this
 x1 -> x1 -> x2 -> x2 - x2 -> x3
p1 -> p2 -> p1 -> p1 -> p3 -> p3

As you can see, node x1 has property p1 at step1 and p2 at step2.
I convert node x1 to embedding and feed it to GCN layer. I also have an embedding for p1 which should be added with the embedding of x1 in the expanded form. Right now my data.x is of the form {x1, x2, x3} and edge index  ([0,0,1,1,1]
[0,1,1,1,3]). So I cant do an embedding(x1)+embedding(p1) as the p's change with every step. How to solve this?

**Comment by rusty1s** [comment, marked as answer]
I am not totally sure I understand the issue, but if your assingment between` x` and `p` changes dynamically, you somehow need to store this assignment somewhere and use it when accessing embeddings, e.g.:
```python
out = self.embedding(x[assign_index[0]]) + self.embedding(p[assign_index[1]])
```
